### PR TITLE
Rewrite few parts of the Logger

### PR DESCRIPTION
**Pull request by iwei20**
Added a getLogLevel() to the Loggable interface so that Levels other than info can be used; default will still be info.  Added methods that can register Loggables so that all of the registered Loggables will log on call, similarly to event registering and firing.  Kept other methods of logging, including logging the Loggable once and manually inputting data to log.

### CODE REVIEW

File: src/com/stuypulse/stuylib/file/FRCLogger.java [hunk 1]
```diff
@@ -1,71 +1,164 @@
...
     public interface Loggable {
-        public String getLog();
+        /**
+         * Implement this method with something that reports
+         * if any data from the class should be fed to the logger
+         * this iteration.
+         * 
+         * @return if any data from the class will be fed to the logger.
+         */
+        public boolean logThisIteration();
```
Comment (Sam-Belliveau) [inline src/com/stuypulse/stuylib/file/FRCLogger.java]: should default to true imo

File: src/com/stuypulse/stuylib/file/FRCLogger.java [hunk 2]
```diff
@@ -1,71 +1,164 @@
...
+     * Registers a new Loggable that upon call will put the next
+     * getLogData() at level getLogLevel() to the log if it will
+     * logThisIteration().
      * 
-     * @param info the information
+     * @param loggable Class implementing Loggable
      */
-    public void logInfo(String info) {
-        mLogger.info(info);
+    public void registerLoggable(Loggable loggable) {
```
Comment (Sam-Belliveau) [inline src/com/stuypulse/stuylib/file/FRCLogger.java]: maybe rename to addLoggable
register seems a bit odd

File: src/com/stuypulse/stuylib/file/FRCLogger.java [hunk 3]
```diff
@@ -1,71 +1,164 @@
...
+     */
+    public void clearLoggables() {
+        loggables.clear();
+    }
+
+    /**
+     * Iterates through all the registered Loggables and logs them with level and data,
+     * if logThisIteration() is true.
+     */
+    public void logRegisteredLoggables() {
```
Comment (Sam-Belliveau) [inline src/com/stuypulse/stuylib/file/FRCLogger.java]: this like a bit of an odd name.

File: src/com/stuypulse/stuylib/file/FRCLogger.java (hunk 2 above)
Comment (Ca7Ac1) [reply src/com/stuypulse/stuylib/file/FRCLogger.java]: Register is to reminisce of events and event handlers for people who come from javascript.  Also it semi makes sense in that context.  Also we want our code to be UNIQUE, not like all those generic adds.

Comment (Ca7Ac1) [reply src/com/stuypulse/stuylib/file/FRCLogger.java]: -Ivan from Ayan's account

File: src/com/stuypulse/stuylib/file/FRCLogger.java (hunk 3 above)
Comment (Ca7Ac1) [reply src/com/stuypulse/stuylib/file/FRCLogger.java]: See above

File: src/com/stuypulse/stuylib/file/FRCLogger.java (hunk 1 above)
Comment (Ca7Ac1) [reply src/com/stuypulse/stuylib/file/FRCLogger.java]: Done

**Pull request merged**